apps/provider-console/src/context/SettingsProvider/SettingsProviderContext.tsx

```diff
@@ -65,83 +65,97 @@ export const SettingsProvider: FC<{ children: ReactNode }> = ({ children }) => {
-        // Use cosmos.directory as a backup if there's no active nodes in the list
-        defaultApiNode = randomNode?.api || "https://rest.cosmos.directory/akash";
-        defaultRpcNode = randomNode?.rpc || "https://rpc.cosmos.directory/akash";
-        selectedNode = randomNode || {
-          api: defaultApiNode,
-          rpc: defaultRpcNode,
-          status: "active",
-          latency: 0,
-          nodeInfo: null,
-          id: "https://rest.cosmos.directory/akash"
-        };
-        updateSettings({ ...settings, apiEndpoint: defaultApiNode, rpcEndpoint: defaultRpcNode, selectedNode, nodes: mappedNodes });
-      } else {
-        defaultApiNode = settings.apiEndpoint;
-        defaultRpcNode = settings.rpcEndpoint;
-        selectedNode = settings.selectedNode;
-        updateSettings({ ...settings, apiEndpoint: defaultApiNode, rpcEndpoint: defaultRpcNode, selectedNode, nodes: mappedNodes });
+            nodeInfo: nodeStatus.nodeInfo,
+            id: customNodeUrl.hostname
+          };
+  
+          updateSettings({ ...settings, apiEndpoint: defaultApiNode, rpcEndpoint: defaultRpcNode, selectedNode, customNode, nodes: mappedNodes });
+        }
+  
+        // If the user has no settings or the selected node is inactive, use the fastest available active node
+        if (!hasSettings || (hasSettings && settings.selectedNode?.status === "inactive")) {
+          const randomNode = getFastestNode(mappedNodes);
+          // Use cosmos.directory as a backup if there's no active nodes in the list
+          defaultApiNode = randomNode?.api || "https://rest.cosmos.directory/akash";
+          defaultRpcNode = randomNode?.rpc || "https://rpc.cosmos.directory/akash";
+          selectedNode = randomNode || {
+            api: defaultApiNode,
+            rpc: defaultRpcNode,
+            status: "active", 
+            latency: 0,
+            nodeInfo: null,
+            id: "https://rest.cosmos.directory/akash"
+          };
+          updateSettings({ ...settings, apiEndpoint: defaultApiNode, rpcEndpoint: defaultRpcNode, selectedNode, nodes: mappedNodes });
+        } else {
+          defaultApiNode = settings.apiEndpoint;
+          defaultRpcNode = settings.rpcEndpoint;
+          selectedNode = settings.selectedNode;
+          updateSettings({ ...settings, apiEndpoint: defaultApiNode, rpcEndpoint: defaultRpcNode, selectedNode, nodes: mappedNodes });
+        }
+      } catch(error) {
+        if (axios.isAxiosError(error)) {
+          console.error('Axios error while fetching nodes:', error.message);
+          if (error.response) {
+            console.error('Server responded with status:', error.response.status);
+            console.error('Server response data:', error.response.data);
+          } else if (error.request) {
+            console.error('No response received. Request details:', error.request);
+          }
+        } else {
+          console.error('Unexpected error:', error);
+        }
```

Comment:
Please undo these changes. This will probably be refactored anyways since the provider console project is still under construction.